commands/bmc.nicole

```diff
@@ -182,24 +180,30 @@ function cmd_datetime_show {
   format_date "${compact}" "${utc}" "$(date +%s)"
 }
 
+function call_netconfig() {
+  local FNAME=${FUNCNAME[1]##cmd_}
+  local PNAME="bmc ${FNAME//_/ }"
+  exec -a "${PNAME}" netconfig --cli "$@"
```

Comment:
`$@` should be without quotes. It is a list of arguments. In quotes it will interpret as a one string with spaces inside.

Reply:
You'd be right if it was `$*`. However `$@` in double quotes prevents split-and-glob on individual arguments. It translates into `"$1" "$2" "$3" ...` whereas `$@` without quotes translates into `$1 $2 $3` and any argument with spaces or globs in it will be split or substituted. We don't want that.

Here is the example:

```
vegman-sx20-2007210001:~$ function some_func() { for a in $@; do echo $a; done }
vegman-sx20-2007210001:~$ some_func abc "def ghi" jkl mno "pqr stu"
abc
def
ghi
jkl
mno
pqr
stu
vegman-sx20-2007210001:~$ function some_func() { for a in "$@"; do echo $a; done }
vegman-sx20-2007210001:~$ some_func abc "def ghi" jkl mno "pqr stu"
abc
def ghi
jkl
mno
pqr stu
vegman-sx20-2007210001:~$ function some_func() { for a in "$*"; do echo $a; done }
vegman-sx20-2007210001:~$ some_func abc "def ghi" jkl mno "pqr stu"
abc def ghi jkl mno pqr stu
```

Reply:
Yes, you are right